### PR TITLE
METAGEN-86

### PR DESCRIPTION
**Pull request by hferentschik**

### CODE REVIEW

File: src/main/java/org/hibernate/jpamodelgen/Context.java [hunk 1]
```diff
@@ -60,7 +60,11 @@
 	private final String persistenceXmlLocation;
 	private final List<String> ormXmlFiles;
 
-	private boolean isPersistenceUnitCompletelyXmlConfigured;
+	/**
+	 * Keeping track which (if any) of the mapping files is xml-mapping-metadata-complete. If all are metadata complete
+	 * no annotation processing will take place.
+	 */
+	private List<Boolean> fullyXmlConfigured;
```
Comment (gunnarmorling) [inline src/main/java/org/hibernate/jpamodelgen/Context.java:67]: I'm wondering whether one still could use a single flag, as we in the end care only whether there is at least one not fully configured file or not. Tried it here: https://github.com/gunnarmorling/hibernate-metamodelgen/commit/3cde76d3cc724ce7099368b240a4d115d2b1425b Tests still pass. WDYT?

Comment (hferentschik) [inline src/main/java/org/hibernate/jpamodelgen/Context.java:67]: Sure. I am not sure why I choose this approach. I think I found it maybe easier to understand when the different "settings" where kept, but you are right, it is not needed for now. Thanks for trying the other approach. Works for me. Just apply it.

This is really quite a hack anyways. Using multiple persistence units with different meta complete settings will most like result in problems anyways. To make this really work one would need to keep the processed classes in buckets per persistence unit. Now we either process all annotated classes or non.

In practice I don't think this is a problem though.

**Comment by gunnarmorling** [comment]
Rebased and applied. Added two more commits for the discussed change and some clean-up.

File: src/main/java/org/hibernate/jpamodelgen/Context.java (hunk 1 above)
Comment (gunnarmorling) [inline src/main/java/org/hibernate/jpamodelgen/Context.java:67]: Ok, I've pushed the change. Is METAGEN-86 really solved with this approach then? Or should we close it and create a new issue for correct approach which considers each PU for itself?

**Comment by hferentschik** [comment]
> Rebased and applied.

Thanks

**Comment by hferentschik** [comment]
> Ok, I've pushed the change. Is METAGEN-86 really solved with this approach then? Or should we close it and create a new issue for correct approach which considers each PU for itself?

I resolved the issue. I don't think we need a follow up issue on this, since imo we cannot implement this 100% correctly.

**Pull request closed without merging**